numba_cuda/numba/cuda/cudadrv/driver.py

```diff
@@ -2626,19 +2686,38 @@ def add_cu_file(self, path):
             cu = f.read()
         self.add_cu(cu, os.path.basename(path))
 
-    def add_file_guess_ext(self, path):
+    def add_file_guess_ext(self, path_or_code):
         """Add a file to the link, guessing its type from its extension."""
-        ext = os.path.splitext(path)[1][1:]
-        if ext == '':
-            raise RuntimeError("Don't know how to link file with no extension")
-        elif ext == 'cu':
-            self.add_cu_file(path)
-        else:
-            kind = FILE_EXTENSION_MAP.get(ext, None)
-            if kind is None:
-                raise RuntimeError("Don't know how to link file with extension "
-                                   f".{ext}")
-            self.add_file(path, kind)
+        if isinstance(path_or_code, str):
+            ext = pathlib.Path(path_or_code).suffix
```

Comment:
Are we missing a bit of logic for handling `.ltoir` here? Or has it gone somewhere else? The original I'm looking at is https://github.com/rapidsai/pynvjitlink/blob/a2f23b7c3c237f2cdde3093c845e0453572503eb/pynvjitlink/patch.py#L170-L171

Reply:
Comment links:
- https://github.com/NVIDIA/numba-cuda/pull/23/files#r1726887376
- https://github.com/NVIDIA/numba-cuda/pull/23/files#r1726884827
- https://github.com/NVIDIA/numba-cuda/pull/23/files#r1726885572